### PR TITLE
Support for security remote_cluster and associated privileges

### PR DESCRIPTION
**Pull request by jakelandis**
This commit adds support for the remote_cluster in the role and role descriptors.
Additionally:
* adds missing references to remote_indices
* add new cluster privilege monitor_stats
* adds related version information where applicable
* updates references to cluster from string[] to proper enumeration

=== 

Note - this PR replaces https://github.com/elastic/elasticsearch-specification/pull/3123 since PR from forks won't pass CI.

### CODE REVIEW

**Comment by jakelandis** [comment]
Follow ups from comments on original PR: 

https://github.com/elastic/elasticsearch-specification/pull/3123#issuecomment-2472759463
> Can you please run make spec-format-fix and reopen this pull request from a branch?

done in https://github.com/elastic/elasticsearch-specification/pull/3125/commits/1d5d03a7c17f78b675e2ae52c52eefc9332f638a

https://github.com/elastic/elasticsearch-specification/pull/3123#discussion_r1840119363
> RemoteClusterPrivilege (enum) -> RemoteClusterPrivileges (class)

done in https://github.com/elastic/elasticsearch-specification/pull/3125/commits/9a70a07e08e1ad76df845a341c40e16721c70720

https://github.com/elastic/elasticsearch-specification/pull/3123#issuecomment-2472759463
> I've opened a pull request to remove it and required a review from you.

Thank you !

https://github.com/elastic/elasticsearch-specification/pull/3123#discussion_r1840343966 and https://github.com/elastic/elasticsearch-specification/pull/3123#discussion_r1840341312
> For me it would be ok to introduce a breaking change by changing the return type to the correct enum type. What about the rest @elastic/devtools-team ?

Still pending...

File: compiler/package-lock.json [hunk 1]
```diff
@@ -33,6 +33,10 @@
         "node": ">=14"
       }
     },
+    "../compiler-rs/compiler-wasm-lib/pkg": {
```
Comment (n1v0lg) [inline compiler/package-lock.json:36]: Hm, I'm not sure we should be touching this file

File: specification/security/_types/Privileges.ts [hunk 2]
```diff
@@ -151,6 +151,10 @@ export enum ClusterPrivilege {
    * @availability stack
    */
   monitor_snapshot,
+  /**
+   * @availability stack
```
Comment (n1v0lg) [inline specification/security/_types/Privileges.ts:155]: Should we also add `since=8.17.0` here?

**Review by n1v0lg — APPROVED** [review]
LGTM from an ES security perspective -- deferring to @flobernd on the final recommendation around whether we want to go from string to enum for privilege values in the read APIs.

File: specification/security/get_builtin_privileges/SecurityGetBuiltinPrivilegesResponse.ts [hunk 3]
```diff
@@ -17,8 +17,16 @@
...
+import {
+  ClusterPrivilege,
+  RemoteClusterPrivilege
+} from '@security/_types/Privileges'
 import { IndexName } from '@_types/common'
 
 export class Response {
-  body: { cluster: string[]; index: IndexName[] }
+  body: {
+    cluster: ClusterPrivilege[]
```
[the remark this answers is not in the document]
Comment (Anaethelion) [reply specification/security/get_builtin_privileges/SecurityGetBuiltinPrivilegesResponse.ts:28]: Enums are treated openly as long as they have the `@non_exhaustive` annotation, which `ClusterPrivilege` and `IndexPrivilege` already have.

[the remark this answers is not in the document]
Comment (pquentin) [reply specification/security/get_builtin_privileges/SecurityGetBuiltinPrivilegesResponse.ts:28]: [ClusterPrivilege is declared as `non_exhaustive`](https://github.com/elastic/elasticsearch-specification/blob/55d431ab4ffe02043ba6f6a41d27f14542f4c792/specification/security/_types/Privileges.ts#L42-L43), which means that in Java and .NET (the only clients that have strongly typed enums) switching from a string to ClusterPrivilege is supported. Receiving unknown values works too.

File: compiler/package-lock.json (hunk 1 above)
Comment (pquentin) [reply compiler/package-lock.json:36]: Yes, it would be nice to revert those unrelated changes (also in package.json).

Comment (jakelandis) [reply compiler/package-lock.json:36]: The PR has a mix of manual changes and automated changes from just building and validation ... should i revert all changes that result from the automation ? (if so, should these be .gitignored instead)

Comment (pquentin) [reply compiler/package-lock.json:36]: Please keep the changes in specification/ (manual) and output/ (generated) only. Thank you!

Comment (jakelandis) [reply compiler/package-lock.json]: done in https://github.com/elastic/elasticsearch-specification/pull/3125/commits/ca540518dd3a8e9c887135179aca043f23252739

File: package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
   "dependencies": {
-    "@stoplight/spectral-cli": "^6.13.1"
+    "@stoplight/spectral-cli": "^6.14.1"
```
[the remark this answers is not in the document]
Comment (jakelandis) [reply package.json:3]: done in https://github.com/elastic/elasticsearch-specification/pull/3125/commits/ca540518dd3a8e9c887135179aca043f23252739

File: specification/security/_types/Privileges.ts (hunk 2 above)
Comment (jakelandis) [reply specification/security/_types/Privileges.ts]: done in https://github.com/elastic/elasticsearch-specification/pull/3125/commits/720e1f0b7cd3f408d799a4d698740056ca37b2f0

**Review by pquentin — COMMENTED** [review]
Thank you for iterating! LGTM.

**Review by pquentin — APPROVED** [review]

**Comment by jakelandis** [comment]
@pquentin / @flobernd - thanks for the guidance. Should I merge this back to the 8.x branch ? The changes here are relevant for 8.17.  (not sure how the branching strategy works for this repo)

**Comment by pquentin** [comment]
Thanks! We use the same branching strategy but backporting is a bit different. I applied the "backport 8.x" label.

**Comment by pquentin** [comment]
Since the automatic backport failed due to a conflict, I created it manually at https://github.com/elastic/elasticsearch-specification/pull/3140 and would appreciate a review.

**Pull request merged**